### ISSUE TITLE
Feign unable to find Spring Cloud Contract LoadBalancer

### ISSUE THREAD

**Issue by rafaelrenanpacheco** [spring-cloud/spring-cloud-contract#1430]
Hello!

When updating Spring Cloud from Hoxton.SR1 to Hoxton.SR5 some consumer tests started to fail. In this update I removed `spring-cloud-starter-netflix-eureka-client` and started to use `spring-cloud-starter-loadbalancer` (along with `spring-cloud-starter-kubernetes` just in case this is of some use for this issue).

After this changes the consumer test started to fail with the following message:

```
No Feign Client for loadBalancing defined. Did you forget to include spring-cloud-starter-netflix-ribbon?
```

After that I tried to use Hoxton.BUILD-SNAPSHOT (which points to Spring Cloud 2.2.4), and the error still occurred.

Based on #1312  this issue shouldn't happen. After some debugging the `SpringCloudLoadBalancerAutoConfiguration` is beaning a `LoadBalancerClientFactory`  and `StubRunnerSpringCloudAutoConfiguration` is beaning a `DiscoveryClient`. But for some reason `FeignClientFactoryBean#loadBalance` isn't finding a load balancer.

This is how I run the consumer test using JUNIT 5 (pseudocode):

```java
@WebMvcTest
@AutoConfigureStubRunner(ids = "com.company:provider:+:stubs:8080")
@ExtendWith(SpringExtension.class)
public class ProviderClientTest {

    @Autowired
    private ProviderClient providerClient;

    @Test
    public void shouldGetSomething() {
        String response = providerClient.getSomething();

        assertThat(response).isEqualTo("response");
    }
}
```

And this is the Feign Client (also pseudocode):

```java
@FeignClient(name = "provider")
public interface ProviderClient {

    @GetMapping("endpoint")
    String getSomething();
}
```

Before that we were using JUNIT 4 and `spring-cloud-starter-netflix-eureka-client`, and it worked just fine.

Is this the proper way to run consumer tests with JUNIT 5 along with the contract loadbalancer?

**Comment by marcingrzejszczak** [comment]
That should work ™️ can you create a small sample that replicates the problem?

**Comment by rafaelrenanpacheco** [comment]
@marcingrzejszczak indeed it should work. Just now I did some tests to ensure that the default configuration should work. For that I removed `@WebMvcTest` and replaced with `@SpringBootTest`, and the tests started to work again. But this is a no go for us because `@SpringBootTest` loads lots of layers (like running flyway migrations on H2 db), and we need that `@WebMvcTest` slice to increase our tests speed.

After some bean hell I found that this context configuration works in a `@WebMvcTest` slice:

```java
@SpringBootConfiguration
@EnableFeignClients
@ImportAutoConfiguration({
        FeignAutoConfiguration.class,
        FeignLoadBalancerAutoConfiguration.class,
        BlockingLoadBalancerClientAutoConfiguration.class})
public class FeignClientConfiguration {
}
```

Now everything is 100% again, thanks!

Best regards,
Rafael Pacheco.
